File: internal/models/game_platform.go
package models

import "time"

// GamePlatform 游戏平台
type GamePlatform struct {
	ID        string              `json:"id" yaml:"id"`               // 平台ID
	Name      string              `json:"name" yaml:"name"`           // 平台名称
	Version   string              `json:"version" yaml:"version"`     // 平台版本
	Type      string              `json:"type" yaml:"type"`           // 平台类型
	OS        string              `json:"os" yaml:"os"`               // 操作系统
	Image     string              `json:"image" yaml:"image"`         // 容器镜像
	Bin       string              `json:"bin" yaml:"bin"`             // 启动路径
	Features  []string            `json:"features" yaml:"features"`   // 平台特性列表
	Config    map[string]string   `json:"config" yaml:"config"`       // 平台配置键值对
	Files     []PlatformFile      `json:"files" yaml:"files"`         // 平台文件
	Installer []PlatformInstaller `json:"installer" yaml:"installer"` // 安装步骤
	CreatedAt time.Time           `json:"created_at"`
	UpdatedAt time.Time           `json:"updated_at"`
}

// PlatformFile 平台文件
type PlatformFile struct {
	ID   string `json:"id" yaml:"id"`     // 文件ID
	Type string `json:"type" yaml:"type"` // 文件类型
	URL  string `json:"url" yaml:"url"`   // 文件URL
}

// PlatformInstaller 安装步骤
type PlatformInstaller struct {
	Move    *InstallerMove    `json:"move,omitempty" yaml:"move,omitempty"`       // 移动文件
	Chmodx  string            `json:"chmodx,omitempty" yaml:"chmodx,omitempty"`   // 添加执行权限
	Extract *InstallerExtract `json:"extract,omitempty" yaml:"extract,omitempty"` // 解压文件
	Command string            `json:"command,omitempty" yaml:"command,omitempty"` // 执行命令
}

// InstallerMove 移动文件配置
type InstallerMove struct {
	Dst string `json:"dst" yaml:"dst"` // 目标路径
	Src string `json:"src" yaml:"src"` // 源文件
}

// InstallerExtract 解压文件配置
type InstallerExtract struct {
	Dst  string `json:"dst" yaml:"dst"`   // 目标路径
	File string `json:"file" yaml:"file"` // 源文件
}

// TableName 返回表名
func (GamePlatform) TableName() string {
	return "game_platforms"
}

File: frontend/src/types/GamePlatform.ts
export type GamePlatformOS = 'Linux' | 'Windows' | 'macOS'
export type GamePlatformStatus = 'active' | 'maintenance' | 'inactive'

export interface GamePlatformFile {
  id: string
  type: string
  url: string
}

export interface GamePlatformInstallerCommand {
  command?: string
  move?: {
    src: string
    dst: string
  }
  chmodx?: string
  extract?: {
    file: string
    dst: string
  }
}

export interface GamePlatformConfig {
  wine?: string
  dxvk?: string
  vkd3d?: string
  python?: string
  proton?: string
  'shader-cache'?: string
  'remote-play'?: string
  broadcast?: string
  mode?: string
  resolution?: string
  wifi?: string
  bluetooth?: string
}

export interface GamePlatform {
  id: string
  name: string
  version: string
  os: GamePlatformOS
  status: GamePlatformStatus
  description: string
  image: string
  bin: string
  data: string
  files: GamePlatformFile[]
  features: string[]
  config: GamePlatformConfig
  installer?: GamePlatformInstallerCommand[]
  createdAt: string
  updatedAt: string
}

export interface GamePlatformQuery {
  page: number
  pageSize: number
  keyword?: string
  status?: string
}

export interface GamePlatformForm {
  id?: string
  name: string
  type: string
  description: string
  status: 'active' | 'inactive' | 'maintenance'
  version: string
  image: string
  bin: string
  data: string
  files: PlatformFile[]
  features: string[]
  config: Record<string, string>
  game?: {
    exe: string
  }
  installer?: {
    command?: string
    move?: {
      src: string
      dst: string
    }
    chmodx?: string
    extract?: {
      file: string
      dst: string
    }
  }[]
} 